subprojects/signing/src/integTest/groovy/org/gradle/plugins/signing/SigningPublicationsIntegrationSpec.groovy

```diff
@@ -347,19 +345,56 @@ class SigningPublicationsIntegrationSpec extends SigningIntegrationSpec {
 
         then:
         executedAndNotSkipped(":publishIvyJavaPublicationToIvyRepository")
+        filesExists(this).forEach { it.assertExists() }
+        filesDoesNotExist(this).forEach { it.assertDoesNotExist() }
 
-        and:
-        ivyRepoFile(jarFileName).assertExists()
-        ivyRepoFile("${jarFileName}.asc").assertExists()
-        ivyRepoFile("ivy-${version}.xml").assertExists()
-        ivyRepoFile("ivy-${version}.xml.asc").assertExists()
-        ivyRepoFile("$artifactId-${version}-source.jar").assertExists()
-        ivyRepoFile("$artifactId-${version}-source.jar.asc").assertExists()
-        ivyRepoFile("$artifactId-${version}.module").assertDoesNotExist()
-        ivyRepoFile("$artifactId-${version}.module.asc").assertDoesNotExist()
 
-        and:
-        outputContains "Publication of Gradle Module Metadata is disabled because you have configured an Ivy repository with a non-standard layout"
+        where:
+        layout     | declaration | filesExists                                       | filesDoesNotExist
+        "standard" | ""          | this.&expectedFilesExistsIvyPublishStandardLayout | this.&expectedFilesDoesNotExistIvyPublishStandardLayout
+        "custom"   | """
+                     patternLayout {
+                       artifact "[artifact]-[revision](-[classifier])(.[ext])"
+                       ivy "[artifact]-[revision](-[classifier])(.[ext])"
+                     }"""
+
+
+                                 | this.&expectedFilesExistsIvyPublishCustomLayout   | this.&expectedFilesDoesNotExistIvyPublishCustomLayout
+    }
+
+    private static List<TestFile> expectedFilesExistsIvyPublishStandardLayout(SigningPublicationsIntegrationSpec spec) {
+        return [
+            spec.ivyRepoFile(spec.jarFileName),
+            spec.ivyRepoFile("${spec.jarFileName}.asc"),
+            spec.ivyRepoFile("ivy-${spec.version}.xml"),
+            spec.ivyRepoFile("ivy-${spec.version}.xml.asc"),
+            spec.ivyRepoFile("$spec.artifactId-${spec.version}-source.jar"),
+            spec.ivyRepoFile("$spec.artifactId-${spec.version}-source.jar.asc"),
+            spec.ivyRepoFile("$spec.artifactId-${spec.version}.module"),
+            spec.ivyRepoFile("$spec.artifactId-${spec.version}.module.asc")
+        ]
+    }
+
+    private static List<TestFile> expectedFilesDoesNotExistIvyPublishStandardLayout(SigningPublicationsIntegrationSpec spec) {
+        return []
```

Comment:
🎨 I would inline this. The column name in the `where` table is enough documentation.